test/unit/pending-balance-test.js

```diff
@@ -0,0 +1,99 @@
+const assert = require('assert')
+const PendingBalanceCalculator = require('../../app/scripts/lib/pending-balance-calculator')
+const MockTxGen = require('../lib/mock-tx-gen')
+const BN = require('ethereumjs-util').BN
+let providerResultStub = {}
+
+const zeroBn = new BN(0)
+const etherBn = new BN(String(1e18))
+const ether = '0x' + etherBn.toString(16)
+
+describe('PendingBalanceCalculator', function () {
+  let balanceCalculator
+
+  describe('#valueFor(tx)', function () {
+    it('returns a BN for a given tx value', function () {
+      const txGen = new MockTxGen()
+      pendingTxs = txGen.generate({
+        status: 'submitted',
+        txParams: {
+          value: ether,
+          gasPrice: '0x0',
+          gas: '0x0',
+        }
+      }, { count: 1 })
+
+      const balanceCalculator = generateBalanceCalcWith([], zeroBn)
+      const result = balanceCalculator.valueFor(pendingTxs[0])
+      assert.equal(result.toString(), etherBn.toString(), 'computes one ether')
+    })
+
+    it('calculates gas costs as well', function () {
+      const txGen = new MockTxGen()
+      pendingTxs = txGen.generate({
+        status: 'submitted',
+        txParams: {
+          value: '0x0',
+          gasPrice: '0x2',
+          gas: '0x3',
+        }
+      }, { count: 1 })
+
+      const balanceCalculator = generateBalanceCalcWith([], zeroBn)
+      const result = balanceCalculator.valueFor(pendingTxs[0])
+      assert.equal(result.toString(), '6', 'computes one ether')
```

Comment:
i dont understand how these values match `'6', 'computes one ether'`

Reply:
This is incorrectly labeled, should be "computes 6 wei of gas".